front/lib/api/assistant/jit_actions.ts

```diff
@@ -104,14 +104,21 @@ async function getJITActions(
         actions.push(action);
       }
 
-      if (
-        filesUsableAsRetrievalQuery.filter((f) => isConversationFileType(f))
-          .length > 0
-      ) {
-        assert(
-          dataSourceView,
-          "No conversation datasource view found for retrieval when trying to get JIT actions"
-        );
+      if (filesUsableAsRetrievalQuery.length > 0) {
+        const dataSources = filesUsableAsRetrievalQuery
+          .filter((f) => isConversationContentNodeType(f))
+          .map((f) => ({
+            workspaceId: auth.getNonNullableWorkspace().sId,
+            dataSourceViewId: f.nodeDataSourceViewId,
+            filter: { parents: null, tags: null },
+          }));
+        if (dataSourceView) {
```

Comment:
Can we rename this one to make it clear it's the conversation data source right?